status/pricing.py

```diff
@@ -20,379 +20,67 @@ def _validate_version_param(self, version):
-                        When not specified, the latest exchange rates will be
-                        used.
-        :param pretty_strings: Output prices as formatted strings instead
-                        of float numbers.
-
-        """
-        exch_rates = self.fetch_exchange_rates(date)
-        all_components = self.fetch_components(component_id, version)
-
-        return_d = all_components.copy()
-
-        for component_id, component in all_components.items():
-            if component['List price']:
-                price, price_per_unit = self._calculate_component_price(component,
-                                                                        exch_rates)
-                if pretty_strings:
-                    price = '{:.2f}'.format(price)
-                    price_per_unit = '{:.2f}'.format(price_per_unit)
-            elif component['Status'] != 'Discontinued':
-                raise ValueError('Empty list price for non-discontinued component')
-            else:
-                price = ''
-                price_per_unit = ''
-
-            return_d[component_id]['price_in_sek'] = price
-            return_d[component_id]['price_per_unit_in_sek'] = price_per_unit
-
-        return return_d
-
-    def get_product_prices(self, product_id, version=None, date=None,
-                           discontinued=False, pretty_strings=False):
-        """Calculate the price for an individual or all products
-
-        :param product_id: The id of the product to calculate price for.
-                        If None, all product prices will be calculated.
-        :param version: The version of product and components to use.
-                        When not specified, the latest version will be used.
-        :param date: The date for which the exchange rate will be fetched.
-                        When not specified, the latest exchange rates will be
-                        used.
-        :param discontinued: If evaluated to False, only products with status
-                        'available' will be included.
-        :param pretty_strings: Output prices as formatted strings instead
-                        of float numbers.
-
-        """
-
-        all_component_prices = self.get_component_prices(version=version, date=date)
-        products = self.fetch_products(product_id, version)
-
-        return_d = products.copy()
-
-        for product_id, product in products.items():
-            if not discontinued and product['Status'] != 'Available':
-                return_d.pop(product_id)
-                continue
+    def _update_last_modified(self, doc):
+        user_name = self.get_current_user().name
+        user_email = self.get_current_user().email
 
-            price_int, price_acad, price_full = self._calculate_product_price(product, all_component_prices)
+        doc['Last modified by user'] = user_name
+        doc['Last modified by user email'] = user_email
```

Comment:
Would it be better to have user and user email as a tuple?

Reply:
A tuple in the json doc? Or do you mean similar to the lock_info where there is a two level thing, so for example: 

```
doc['Last modified'] = {'user': user_name, 'email': user_email}
```
?

I would agree that that would be better, but I'm not sure it's big enough of an improvement to be worth the change?

Reply:
The draft document is huge anyway, so I guess a few more lines won't make a difference.